building/glass/upload.py

```diff
@@ -123,20 +119,47 @@ def gs_rsync(local_path: str, remote_path: str, boto_path: str):
 """.lstrip()
 
 
-def perform_uploads(uploads: dict, branch: str) -> None:
+def upload_gs(staging, root: str, branch_config: aptbranch.Config):
+    branch = branch_config.name
+    if not GS_BRANCH_REGEX.match(branch):
+        raise Exception("not an uploadable branch: %s" % branch)
     if not os.path.exists(BOTO_PATH):
         raise Exception("you need to put the GCP service account private key file into /homeworld/boto-key")
+    botoconfig_name = os.path.join(staging, "boto.config")
+    with open(botoconfig_name, "w") as bout:
+        with open(BOTO_PATH, "r") as f:
+            project_id = json.load(f)["project_id"]
+        bout.write(BOTO_TEMPLATE % (BOTO_PATH, project_id))
+        bout.flush()
+    gs_rsync(root, branch, botoconfig_name)
+
+
+def upload_rsync(staging, root: str, branch_config: aptbranch.Config):
+    branch_name_parts = branch_config.name.split('/', 1)
+    target_dir = os.path.join(branch_config.upload_config["dir"], branch_name_parts[1])
```

Comment:
Maybe use unpacking, instead of branch_name_parts?

i.e. `host, path = branch_config.name.split('/', 1)`